### ISSUE TITLE
Replace All does not respect 'In selection'; stops if it causes syntax error

### ISSUE THREAD

**Issue by fafalone** [twinbasic/twinbasic#1023]
**Describe the bug**
See title.

**To Reproduce**
Highlight a large selection of code, select 'Find in selection', attempt to replace all.

**Expected behavior**
Replace all should only do so in selected text when 'Find in selection' is selected. 
It should not stop if an operation causes an error. Multi-step replace operations may have invalid syntax as an intermediate.

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Windows 10
 - twinBASIC compiler version 0.15.82
 - 
**Additional context**

**Comment by mburns08109** [comment]
> It should not stop if an operation causes an error. Multi-step replace operations may have invalid syntax as an intermediate.

On the other hand, it can help you prevent undesired/accidental outcomes.
Perhaps the user could have the option to select "Ignore Errors / Stop on syntax errors"?

**Comment by fafalone** [comment]
More options are always better, but if there are syntax errors, undo is only a click or keystroke away.

**Comment by mburns08109** [comment]
Yeah, I hear you - what'd be really good would be a "selective Undo" capability (rather than the "serial Undo" we see now). (I'm thinking something like a "global find & replace" function that you work with the keyboard hotkeys doing Find Next/Replace -and/or- Find Next/(Skip)Find Next - only in the reverse semantic - Find Undo/Undo -and/or- Find Undo/(skip)Find Undo.)
Yeah - that'd be the better answer.

**Comment by bclothier** [comment]
Just as FYI - VSCode has that quirk, too. I think that's just how it's implemented in the Monaco editor. Awkward, though.
